src/MSXMotherBoard.cc

```diff
@@ -1018,36 +1037,55 @@ void RemoveExtCmd::tabCompletion(std::vector<string>& tokens) const
 StoreSetupCmd::StoreSetupCmd(MSXMotherBoard& motherBoard_)
 	: Command(motherBoard_.getCommandController(), "store_setup")
 	, motherBoard(motherBoard_)
+	, depthOptionMap({
+		{ "none", MSXMotherBoard::SetupDepth::NONE },
+		{ "machine_only", MSXMotherBoard::SetupDepth::MACHINE_ONLY },
+		{ "with_extensions", MSXMotherBoard::SetupDepth::WITH_EXTENSIONS },
+		{ "with_extensions_and_pluggables", MSXMotherBoard::SetupDepth::WITH_EXTENSIONS_AND_PLUGGABLES },
+		{ "with_extensions_and_pluggables_and_media", MSXMotherBoard::SetupDepth::WITH_EXTENSIONS_AND_PLUGGABLES_AND_MEDIA },
+		{ "complete_state", MSXMotherBoard::SetupDepth::COMPLETE_STATE }
+	})
 {
 }
 
 void StoreSetupCmd::execute(std::span<const TclObject> tokens, TclObject& result)
 {
-	checkNumArgs(tokens, Between{1, 2}, Prefix{1}, "?filename?");
+	checkNumArgs(tokens, Between{2, 3}, Prefix{1}, "depth ?filename?");
+
+	std::string depthArg = std::string(tokens[1].getString());
+	if (depthOptionMap.count(depthArg) == 0) {
```

Comment:
In older c++ versions this was indeed the way to do this. But nowadays you can also write:
```if (!optionMap.contains(level)) { ...```
However, immediately after you do the same search again via `optionMap.at(...)` (even multiple times). Better to do only one search:
```
auto it = optionMap.find(level);
if (it == optionMap.end()) { throw ... }
... use it->second
```

BTW: do you think `depth` is a better name than `level`? I mean to communicate to the user. Personally I think not.

Reply:
When I was making it, I did think that `depth` is better than `level`, as it is a more precise term. Especially if you see it as a hierarchical thing (which it is).

I'll make the other changes.

Reply:
@m9710797 do you still want me to change this code? If not, please close the thread.

Reply:
I fixed it now, let me know if you find one I missed.